### PR TITLE
internal solution for cf-1520F1

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/silver/cf-1520F1.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+**Time Complexity:** $\mathcal{O}(\log N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int f(int r, int k) {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1520F1.mdx:33]: better name

Comment (freakin23) [reply solutions/silver/cf-1520F1.mdx]: done

**Review by SansPapyrus683 — APPROVED** [review]
show screenshots of the _new solutions_ ACing

**Comment by freakin23** [comment]
> show screenshots of the _new solutions_ ACing

Screenshot from 2024-06-25 19-34-28
Screenshot from 2024-06-25 19-33-53

**Review by envyaims — APPROVED** [review]
asifbvaiksfgoaugfho;uaghbwsGWOPIHEGOHEGROUEQRGO;UHO;AGHROAHRDOI;HOP'opihioph'OHOIHOIHOI'HOIHh'h'oihoi'dfoig'ahge'oeroighaeiowhgio'aehgioheargopihepohgiop'ehragopheaoprgheroaighoperhehagp'haeghpierpieagpieahpihgoebojewhfouwheRGP'hwipoeghpWI'GPWIEPHI'WHGIPHWEpipi'wGP'IHGPI'WEIp'pw'igp'iwGIPWHEWGP'EHIP'GUFIWIGPUPS'DIUGPISHSPIHOSDHSHGPSDIOGHIP'hp'hip'hip'hgpishgip'shgpid'

**Review by ryanchou-dev — APPROVED** [review]

**Comment by freakin23** [comment]
> asifbvaiksfgoaugfho;uaghbwsGWOPIHEGOHEGROUEQRGO;UHO;AGHROAHRDOI;HOP'opihioph'OHOIHOIHOI'HOIHh'h'oihoi'dfoig'ahge'oeroighaeiowhgio'aehgioheargopihepohgiop'ehragopheaoprgheroaighoperhehagp'haeghpierpieagpieahpihgoebojewhfouwheRGP'hwipoeghpWI'GPWIEPHI'WHGIPHWEpipi'wGP'IHGPI'WEIp'pw'igp'iwGIPWHEWGP'EHIP'GUFIWIGPUPS'DIUGPISHSPIHOSDHSHGPSDIOGHIP'hp'hip'hip'hgpishgip'shgpid'

👀

**Pull request merged**